api/src/main/java/jakarta/enterprise/inject/build/compatible/spi/InterceptorInfo.java

```diff
@@ -0,0 +1,50 @@
+package jakarta.enterprise.inject.build.compatible.spi;
+
+import jakarta.enterprise.inject.spi.InterceptionType;
+import jakarta.enterprise.lang.model.AnnotationInfo;
+
+import java.util.Collection;
+
+/**
+ * Interceptors are managed beans annotated {@link jakarta.interceptor.Interceptor @Interceptor}.
+ * An interceptor declares a set of {@linkplain jakarta.interceptor.InterceptorBinding interceptor binding annotations},
+ * used to associate the interceptor with target beans. It also declares at least one interceptor method:
+ * <ul>
+ * <li>for business method invocations: {@link jakarta.interceptor.AroundInvoke @AroundInvoke},</li>
+ * <li>for constructor invocations: {@link jakarta.interceptor.AroundConstruct @AroundConstruct},</li>
+ * <li>for lifecycle callbacks: {@link jakarta.annotation.PostConstruct @PostConstruct},
+ * {@link jakarta.annotation.PreDestroy @PreDestroy}.</li>
+ * </ul>
+ * Finally, an interceptor that is <em>enabled</em> also declares {@link jakarta.annotation.Priority @Priority}.
```

Comment:
@manovotn Do you think this description makes sense? I wrote it from the CDI and especially CDI Lite perspective, knowing that interceptors don't always have to be beans, but they kinda have to in CDI Lite. So I'm wondering a little if this may be confusing.

Reply:
@Ladicek I think it is mostly fine. Interceptors are beans and for purposes of Lite, your description fits. Just two minor notes:

> It also declares at least one interceptor method:

`It also declares at most one of each interceptor methods:` 
(as declaring same interception method multiple times should be considered an error)

> Finally, an interceptor that is <em>enabled</em> also declares {@link jakarta.annotation.Priority @Priority}.

`Finally, an interceptor is by default disabled. In order to enable it, it should declare {@link jakarta.annotation.Priority @Priority}.` 
(just to state what the default is)

Reply:
Done.

I vaguely recall some spec saying that an interceptor must declare at least one interceptor method, but I can't find that anymore :-) So I went with "at most one interceptor method for each interception type".

I reworded the sentence about `@Priority` to better convey what you suggest, but still fit into the overall text flow.

I also added a note that this description applies to Lite, and there are more ways to declare an interceptor in Full.